File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    res = []
    
    for i in s:
        if i == '(':
            res.append(i)
        else: # 만약 ( 아니라면 두가지 경우를 확인하는데 비어 있는 경우에는 false, ) 일 경우에는 pop
            if res == []:
                return False
            else :
                res.pop()
            
    return res == []  # res가 [](빈 배열)과 동일하다면? True 반환, res 안에 문자가 있다면 동일하지 않기 때문에 false 반환

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let res = []
    
    for(let i = 0; i < s.length; i++){
        if(s[i] === "("){
            res.push(s[i])
        } else {
            if(res.length === 0){
                return false
            } else {
                res.pop()
            }
        } 
    }
    return res.length === 0 ? true : false
}
